AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -159,22 +171,50 @@ - (void)_layoutNodes:(NSArray *)nodes ofKind:(NSString *)kind atIndexPaths:(NSAr
   if (!nodes.count) {
     return;
   }
-  
+
+  NSUInteger nodeCount = nodes.count;
+  NSMutableArray<ASCellNode *> *allocatedNodes = [NSMutableArray arrayWithCapacity:nodeCount];
   dispatch_group_t layoutGroup = dispatch_group_create();
-  ASSizeRange *nodeBoundSizes = (ASSizeRange *)malloc(sizeof(ASSizeRange) * nodes.count);
+  ASSizeRange *nodeBoundSizes = (ASSizeRange *)malloc(sizeof(ASSizeRange) * nodeCount);
+  BOOL isMainThread = [NSThread isMainThread];
```

Comment:
I need to go around the framework to enforce adoption, as it is inconsistent at the moment, but probably worth using ASDisplayNodeThreadIsMain() here.  We may want to rename that too, maybe ASOnMainThread() or something.